### DISCUSSION TITLE
Questions for challenge 2

### DISCUSSION THREAD

**Discussion by nhat-m-le** [surgical-robotics-ai/surgical_robotics_challenge#44] [Q&A] [answered]
Hi, 

I have 2 questions:
1) Are we allowed to use additional cameras in the environment for visual feedback purpose in challenge 2?
2) For the docker to be submitted, is there any instruction/requirements on the structures and what must be included? There are quite a few changes to the official challenge repo over the last month, would we need to make sure our docker runs/includes the latest version?

Thanks, 
Nhat

**Comment by adnanmunawar** [comment, marked as answer]
Hello Nhat,
Thanks for your questions. Here are the answers to them:

>Are we allowed to use additional cameras in the environment for visual feedback purpose in challenge 2?

For developing your algorithms, you can use additional cameras but not for the final evaluation. You can of course move the camera (ECM) to get different viewpoints during the final evaluation.

> For the docker to be submitted, is there any instruction/requirements on the structures and what must be included?

Yes, we are going to upload the instructions for submitting your docker containers shortly. From a code structure point of view, we recommend keeping your code outside of the surgical_robotics_challenge assets folders (everything that is part of this repo) as we are going to reset them to get rid of any user modifications.  This information can be found here (https://collaborative-robotics.github.io/surgical-robotics-challenge/system-setup.html). 

> There are quite a few changes to the official challenge repo over the last month, would we need to make sure our docker runs/includes the latest version?

Almost all of the changes are internal improvements/feature additions that should not affect the interfaces for performing the tasks. However, yes, it's a good idea to pull all the updates. On the same note, it's also recommended to update AMBF. For the evaluation, we are going to reset the surgical_robotics_challenge assets to the latest version in your submitted containers and may rename non-interface methods and variables so that a team only uses the provided interfaces to complete the challenge.
